main/design_documents/__ARM_FEATURE_MOPS.rst

```diff
@@ -0,0 +1,75 @@
+=============================================================
+Introducing extensions for memcpy family of memory operations
+=============================================================
+
+Following the announcement of the `2021 Architecture Extensions
+<https://community.arm.com/developer/ip-products/processors/b/processors-ip-blog/posts/arm-a-profile-architecture-developments-2021>`_,
+we are proposing these aditions to the ACLE to support the new instructions
+that standardise the memcpy family of operations.
+
+Here's the rationale behind the contents:
+
+Feature test macro
+##################
+
+In order for users to determine the availability of the new memory operation
+instructions, as well of new potential intrinsics related to them, we are
+proposing the addition of the following feature test macro, to be defined to 1
+if the new extension is available:
+
+::
+
+  __ARM_FEATURE_MOPS
+
+
+Intrinsics
+##########
+
+Given that the purpose of the new instructions is not only optimizing but also
+standardising the software implementation of memcpy, memmove and memset
+operations, the general use of these instructions should be covered by the code
+generation of compilers and library implementations. Therefore, we feel there
+might be no motivation to introduce new intrinsics covering this scenario.
+
+The new extension also introduces a few variations of such instructions covering
```

Comment:
```suggestion
Armv8.8-A and Armv9.3-A also introduce a few variations of such instructions covering
```

Also, what do you mean with `such instructions`? memcpy, memmove and memset? If so, these are better referred as functions, not instructions?